app/src/main/java/at/roteskreuz/stopcorona/model/repositories/QuarantineRepository.kt

```diff
@@ -255,70 +252,80 @@ class QuarantineRepositoryImpl(
         )
     }.subscribeOnComputation()
         .debounce(50, TimeUnit.MILLISECONDS) // some of the shared prefs can be changed together
-        .map {
-            it.mapToQuarantineStatus()
+        .map { prerequisites ->
+            prerequisites.toQuarantineStatus()
         }
         // don't notify again if nothing has changed
-        .distinctUntilChanged { oldState, newState ->
-            // display end quarantine info if quarantine has ended
+        .distinctUntilChanged()
+
+    init {
```

Comment:
Please also update 
```
    single<QuarantineRepository>(createOnStart = true) {
        QuarantineRepositoryImpl(
            appDispatchers = get(),
            preferences = get(),
            configurationRepository = get(),
            workManager = get()
        )
    }
```

Reply:
No need for createOnStart. All state that is listened to in init can only be changed throught this repository.
So init alone will gaurantee, that we arre listening to the changes before the are made.

Reply:
Currently, no quarantine update will be done unless someone will use QuarantineRepository. I guess we want to have quarantine state updated as soon as the application is started and not wait until a special screen or use case when quarantineRepository is injected.

Reply:
Decide as you want.